src/Acuminator/Acuminator.Utilities/DiagnosticSuppression/SuppressionManager.cs

```diff
@@ -185,85 +212,63 @@ private static SuppressionDiffResult CompareFiles(SuppressionFile oldFile, Suppr
-			if (rootNode == null)
+			if (assembly == null)
 			{
-				return (null, default);
+				return false;
 			}
 
-			var diagnosticNode = rootNode.FindNode(diagnostic.Location.SourceSpan);
-			if (diagnosticNode == null)
-			{
-				return (null, default);
-			}
+			var file = _fileByAssembly.GetOrAdd(assembly, (SuppressionFile)null);
 
-			var targetNode = FindTargetNode(diagnosticNode);
-			if (targetNode == null)
+			if (file == null)
 			{
-				return (null, default);
+				return false;
 			}
 
-			var targetSymbol = semanticModel.GetDeclaredSymbol(targetNode, cancellation);
-			if (targetSymbol == null)
+			if (file.GenerateSuppressionBase)
 			{
-				return (null, default);
+				if (IsSuppressableSeverity(diagnostic?.Descriptor.DefaultSeverity))
+				{
+					file.AddMessage(message);
+				}
+
+				return true;
 			}
 
-			var assemblyName = targetSymbol.ContainingAssembly?.Name;
-			if (string.IsNullOrEmpty(assemblyName))
+			if (!file.ContainsMessage(message))
 			{
-				return (null, default);
+				return false;
 			}
 
-			var id = diagnostic.Id;
-			var target = targetSymbol.ToDisplayString();
+			return true;
+		}
+
+		private static bool IsSuppressableSeverity(DiagnosticSeverity? diagnosticSeverity) =>
+			diagnosticSeverity == DiagnosticSeverity.Error || diagnosticSeverity == DiagnosticSeverity.Warning;
 
-			// Try to obtain token in case of member declaration syntax as we do not want to store the text
-			// of the entire declaration node
-			var token = default(SyntaxToken?);
-			if (diagnosticNode is MemberDeclarationSyntax memberDeclaration)
+		private static void CheckIfInstanceIsInitialized()
+		{
+			if (Instance == null)
 			{
-				try
-				{
-					token = memberDeclaration.FindToken(diagnostic.Location.SourceSpan.Start);
-				}
-				catch (ArgumentOutOfRangeException)
-				{
-					token = null;
-				}
+				throw new InvalidOperationException($"{nameof(SuppressionManager)} instance was not initialized");
 			}
-
-			var syntaxNode = token != null ?
-				token.ToString() :
-				// Replace \r symbol as XDocument does not preserve it in suppression file
-				diagnosticNode.ToString().Replace("\r", "");
-			var message = new SuppressMessage(id, target, syntaxNode);
-
-			return (assemblyName, message);
 		}
 
-		private static SyntaxNode FindTargetNode(SyntaxNode node)
+		private static string GetXDocumentStringWithDeclaration(System.Xml.Linq.XDocument xDocument)
 		{
-			var targetNode = node
-				.AncestorsAndSelf()
-				.Where(a => _targetKinds.Contains(a.Kind()))
-				.FirstOrDefault();
-
-			// Use first variable in case of field declaration as it may contain multiple variables and therefore
-			// cannot be used to obtain declared symbol
-			return targetNode is FieldDeclarationSyntax fieldDeclaration ?
-				fieldDeclaration.Declaration?.Variables[0] :
-				targetNode;
+			var builder = new System.Text.StringBuilder(capacity: 65);
+
+			using (TextWriter writer = new Utf8StringWriter(builder))
```

Comment:
The class Utf8StringWriter is used only in this place. Maybe it is better to remove it?

Reply:
The problem solved here is that that it's hard to get xml string with a xml declaration from XDocument. You need to use code like this to get it. And the string writer doesn't allow you to set custom encoding for string writer. You need to created derived class to do it. And its default one is UTF-16.
 That's why `Utf8StringWriter` is created. You need a class derived from string writer.

Reply:
And why can't we use UTF-16?

Reply:
The standard generated xml file with `xdocument.Save(path)` is UTF-8. I think it is not good if sometimes the suppression file will have UTF-8 and other times - UTF-16. I'd like to stick to the default implementation provided by the XDocument